In [1]:
import fakenewsutilities as fns
import pandas as pd
from sklearn.model_selection import train_test_split

###  1. read the data, then clean the data

In [2]:
df = pd.read_csv('tweets_labeled.csv')

In [3]:
df_clean = fns.wash_pandas_str(df)

In [4]:
df_clean.head()

tweet_id                                               text  \
0  1161040537207463936  The Endangered Species Act saved the bald eagl...   
1  1176360756239118342  Interesting concept -- impeach first, find fac...   
2  1099036648573145088                      #BuildTheWall #DeportThemAll    
3  1092915693203480577  Why would the MEXICAN GOV’T fund this? Who are...   
4  1149038450668187654  Sweden Announces Plan To Get 100% Of Energy Fr...   

   label  
0      1  
1      1  
2      0  
3      0  
4      0

### 2. Training the model

In [12]:
# get X and Y and split the data.
X = df_clean.iloc[:,0:2]
Y = df_clean.iloc[:,2]

X_train, X_test, Y_train, Y_test = train_test_split(X, Y, random_state = 45, test_size  = 0.3)

train_df, fake_prob_prior = fns.naive_bayes_train(X_train, Y_train)

### 3. Evaluating the model

In [13]:
Word_frequency_feature = fns.naive_bayes_generate_feature(train_df, fake_prob_prior, X_test, Y_test)

1000 tested, accuracy 0.889000
2000 tested, accuracy 0.890000
3000 tested, accuracy 0.888667
4000 tested, accuracy 0.887750
5000 tested, accuracy 0.885600
6000 tested, accuracy 0.884500
7000 tested, accuracy 0.885000
8000 tested, accuracy 0.884000
9000 tested, accuracy 0.884000
10000 tested, accuracy 0.883700
11000 tested, accuracy 0.884273
12000 tested, accuracy 0.885083
13000 tested, accuracy 0.886385
14000 tested, accuracy 0.886429
15000 tested, accuracy 0.886600
16000 tested, accuracy 0.886938
17000 tested, accuracy 0.885765
18000 tested, accuracy 0.886111
19000 tested, accuracy 0.886421
20000 tested, accuracy 0.886500
21000 tested, accuracy 0.886143
22000 tested, accuracy 0.886045
23000 tested, accuracy 0.885652
24000 tested, accuracy 0.884875
25000 tested, accuracy 0.885440
26000 tested, accuracy 0.885038
27000 tested, accuracy 0.884926
28000 tested, accuracy 0.885179
29000 tested, accuracy 0.885172
30000 tested, accuracy 0.885567
31000 tested, accuracy 0.885806
32000 tested, acc

Because the test takes a few mins to run, I add a print() in the function to print the processin and accuracy every 1000 tweets tested. 

In this time, I used the most high frequency 500 words as feature, accuracy is more than 88%

As I tried in word_frequency_naivebayes.ipynb, the accuracy of use all words as feature is 96.6%.

If we use different numbers of high-frequency words like 800 or 1000, the accuracy will also be higher.